### ISSUE TITLE
401 logged to console despite successful request to solidcommunity.net server

### ISSUE THREAD

**Issue by AJamesPhillips** [inrupt/solid-client-js#1216]
### Search terms you've used

401

### Bug description

I can successfully use `getSolidDataset` to get a directory but it also logs a 401 error to the console.

### To Reproduce

1. Call `getSolidDataset`

**Minimal reproduction**

https://github.com/AJamesPhillips/solid-bugs/tree/1567be3

(I decided not to use [codesandbox](https://codesandbox.io/s/github/inrupt/solid-client-js/tree/main/.codesandbox/sandbox) as there were a lot of error and warning messages to the console).

### Expected result

No error log out.

### Actual result

![Solid_Bug](https://user-images.githubusercontent.com/526509/130146636-696b9d1d-3723-437a-844d-ec100c6aea5b.png)

### Environment

```
$ npx envinfo --system --npmPackages --binaries --npmGlobalPackages --browsers

  Browsers:
    Brave Browser: 92.1.27.109
    Chrome: 92.0.4515.159
  npmPackages:
    @inrupt/solid-client: ^1.1.0 => 1.11.0 
    @inrupt/solid-client-authn-browser: ^1.11.0 => 1.11.0 
    @inrupt/vocab-common-rdf: ^0.6.3 => 0.6.3 
    buffer: ^6.0.2 => 6.0.3 
    clean-webpack-plugin: ^3.0.0 => 3.0.0 
    html-webpack-plugin: ^4.5.0 => 4.5.2 
    lodash: ^4.17.20 => 4.17.21 
    rdf-namespaces: ^1.9.2 => 1.9.2 
    solid-auth-client: ^2.5.3 => 2.5.6 
    tripledoc: ^4.4.0 => 4.4.0 
    webpack: ^5.5.0 => 5.51.1 
    webpack-cli: ^4.2.0 => 4.8.0 
    webpack-dev-server: ^3.11.0 => 3.11.2 
  npmGlobalPackages:
    npm-run: 5.0.1
    npm: 6.14.8
    pnpm: 6.3.0
```

**Comment by NSeydoux** [comment]
Thanks for reporting this ! I'll see where that might come from. And no problem on not using codesandbox, any way of reproducing is fine :)

**Comment by NSeydoux** [comment]
TL;DR: That's a normal behaviour, and it will disappear if you add trailing slashes to container IRIs.

If you want the detailed version, it looks like what is happening is the following: 
- You try to make an authenticated GET to `https://some.pod/container`. Note that the container URL does not include a trailing slash, which it should.
- The Solid server redirects your request to `https://some.pod/container/` (trailing slash included) because that's the canonical IRI of the resource
- At this point, you have an authentication issue: the [Solid-OIDC spec](https://solid.github.io/solid-oidc/#tokens) mandates that the Access Tokens (which are used to prove that a given application has access to certain resources on a Pod) uses a security mechanism called DPoP. Among other things, this mechanism ensures that the token used to authenticate a request is scoped to the target resource, in order to prevent extraction. That means that if you're giving an access token to an evil Pod Server serving `https://evil.pod/your-resource`, it is not going to be able to use the token to access `https://another.pod/your-banking-data`. However, it also means that the access token used for `https://some.pod/container` isn't valid for `https://some.pod/container/`, because the tagret IRIs don't match (even just by a trailing slash). That's what gets you a 401.
- The authentication library that you are using, namely `@inrupt/solid-client-authn-browser`, knows that this can be an issue. On 401, it checks that the response IRI matches the IRI which was initially fetched (scripts aren't aware of redirections for security reasons, they don't see the intermediary 3XX responses). If it gets a 401 and the IRIs don't match, it knows it should re-attempt with the redirected target IRI (to construct an appropriate authentication header). If you're interested about the details for this, the actual implementation is available at https://github.com/inrupt/solid-client-authn-js/blob/a48f1cccdc7ed82d5f56f2738f97e0a2d8724ed7/packages/core/src/authenticatedFetch/fetchFactory.ts#L275.
- The request to the container with the canonical IRI is not redirected, so its authentication is valid, and the app gets the correct result without being aware of the 401 that was basically intercepted by the authenticated fetch library.

You can test this by manually adding a trailing slash to the container IRI at https://github.com/AJamesPhillips/solid-bugs/blob/1567be367810cd4e3efa5a0a55f4037ee5beb7c6/src/index.js#L113, and see that the 401 is no longer logged.

**Comment by AJamesPhillips** [comment]
@NSeydoux that is an amazing response.  Thank you.

> anything we could have done at a library level to improve the developer experience ?

Add a line of documentation?
